src/shmem.c

```diff
@@ -1471,3 +1472,82 @@ void print_recycle_list_fullness(void)
 	log_info("  Domains: %u/%u (%.2f%%)", recycler->domain.count, RECYCLE_ARRAY_LEN, (double)recycler->domain.count / RECYCLE_ARRAY_LEN * 100.0);
 	log_info("  DNS Cache: %u/%u (%.2f%%)", recycler->dns_cache.count, RECYCLE_ARRAY_LEN, (double)recycler->dns_cache.count / RECYCLE_ARRAY_LEN * 100.0);
 }
+
+/**
+ * @brief Dumps the string table to a temporary file.
```

Comment:
What kind of strings are dumped here? Do we need to make sure that no privacy related information are dumped?

Reply:
All strings FTL knows about. Mainly domains, host names, and IP addresses. But it can also be group strings (like `0,1,2,3`) or local interface names or MAC addresses (if `clients` are configured by these).

Debugging is not really concerned about privacy levels, e.g., `debug.queries` also always prints the full strings as FTL needs them internally to handle to queries correctly. Only API responses and database records are affected by privacy levels.